packages/api-docs/src/PageJsonGenerator.types.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * Options that define the information necessary to find the api.json file
+ * and create individual page json files out of it
+ */
+export interface IPageJsonOptions {
+  apiJsonPaths: string[];
+  pageJsonFolderPath: string;
+  pageNames: string[];
+  kind: PageKind;
+}
+
+/**
+ * The name of the page and what kind of page it is
+ */
+export interface IPage {
+  pageName: string;
+  kind: PageKind;
+}
+
+/**
+ * Used to keep track of where the page will live on the site
+ */
+export enum PageKind {
+  References = 'References',
+  Components = 'Components'
+}
+
+export interface ITokenJson {
+  text: string;
+  hyperlinkedPage?: string;
+  pageKind?: PageKind;
+}
+
+export interface ITableRowJson {
+  name: string;
+  kind?: 'Method' | 'Property';
+  typeTokens: ITokenJson[];
+  defaultValue?: string;
+  descriptionHtml: string;
+  deprecated: boolean;
+  deprecatedMessage?: string;
+}
+
+export interface IEnumTableRowJson {
+  name: string;
```

Comment:
A few of these interface members are common. You might consider them sharing a common base via extends or `Pick<T. [key of T]`